packages/sanity/src/core/form/inputs/arrays/ArrayOfObjectsInput/ArrayOfObjectsFunctions.tsx

```diff
@@ -93,19 +94,12 @@ export function ArrayOfObjectsFunctions<
           menu={
             <Menu>
               {schemaType.of.map((memberDef, i) => {
-                // Use reference icon if reference is to one schemaType only
-                const referenceIcon =
-                  isReferenceSchemaType(memberDef) &&
-                  (memberDef.to || []).length === 1 &&
-                  memberDef.to[0].icon
-
-                const icon = memberDef.icon || memberDef.type?.icon || referenceIcon
                 return (
                   <MenuItem
                     key={i}
```

Comment:
I notice that we could use the `memberDef.type` here instead of the array index. Maybe a worthy small refactor to do in the same PR?